src/controllers/trainee/Controller.ts

```diff
@@ -16,109 +16,91 @@ class TraineeController {
   }
   private constructor() { }
 
-  create = async (req: IRequest, res: Response): Promise<void> => {
+  create = async (req: IRequest, res: Response): Promise<Response> => {
     try {
       const userId = req.user.userId;
       const data = req.body;
       const record = { ...data, userId };
       const result = await this.userRepo.create(record);
       if (result) {
         delete result._id;
-        this.systemResponse.success(
+        return this.systemResponse.success(
           res,
           `Trainee added Successfully`,
           200,
           result
         );
-      } else {
-        this.systemResponse.success(res, `Unauthorized User`, 403, result);
       }
+      return this.systemResponse.success(res, `Unauthorized User`, 403, result);
     } catch (err) {
-      this.systemResponse.failure(res, err.message, 500, err);
+      return this.systemResponse.failure(res, err.message, 500, err);
     }
   };
 
-  list = async (req: IRequest, res: Response) => {
+  list = async (req: IRequest, res: Response): Promise<Response> => {
     try {
       const { skip, limit, sort, search, ...query } = req.query;
       const options: object = { skip, limit, sort };
-      const filter: object = await this.initSearch(search);
+      const filter: object = await initSearch(search);
       let result;
-      if (Object.keys(filter).length) {
-        result = await this.userRepo.getAllRecord(filter, options);
-      } else {
-        result = await this.userRepo.getAllRecord(query, options);
-      }
+      const querySchema = Object.keys(filter).length !== 0 ? filter : query;
```

Comment:
Object.keys(filter).length !== 0 can be written as Object.keys(filter).length

Reply:
Done